apiTools/playerDocRequest.js

```diff
@@ -13,7 +13,7 @@ const getDoc = async (tag, options={}) => {
         if(tag.length===32) doc = await Player.findById(tag);
         else {
             doc = await Player.findOne({ nameLower: tag.toLowerCase() });
-            if(!doc && /#\d{4}$/.test(tag)) doc = await Player.findOne({ discord: tag });
+            if(!doc && /#\d{4}$/.test(tag)) doc = await Player.findOne({ "doc.discord": tag });
```

Comment:
How does this actually fix anything? `discord` field is at the root level
https://github.com/PitPanda/PitPandaProduction/blob/e8b826039fa6e2eedf7bf8a7d661af95a238e5e2/models/Player/index.js#L16

Reply:
Looked at the relevant code and you're totally right my bad. I was trying to figure out why a request like [https://pitpanda.rocks/api/playerDoc/jojo-#4277](https://pitpanda.rocks/api/playerDoc/jojo-#4277) doesn't work and I think it's due to the `#` character present in all Discord tags as that character prevents the rest of the URL from being sent to the server ([https://en.wikipedia.org/wiki/Fragment_identifier](https://en.wikipedia.org/wiki/Fragment_identifier)), so instead of searching for a Discord matching `jojo-#4277` it just searches for `jojo-` and fails. Not sure which way this should be fixed, perhaps a stand-in character for the hash character.

Reply:
`encodeURIComponent` the hashtag (becomes `%23`): https://pitpanda.rocks/api/playerDoc/jojo-%234277

Reply:
Ahh I see that works fine. I'll close this